include/tvm/target/target_kind.h

```diff
@@ -57,6 +52,8 @@ class TargetKindNode : public Object {
   int device_type;
   /*! \brief Default keys of the target */
   Array<String> default_keys;
+  /*! \brief Function used to preprocess on target creation */
+  PackedFunc preprocessor;
```

Comment:
Is it a common option? We can also make it available via set_attr instead

Reply:
Yes, it is a common option. If present, it is called every time on target creation to pre-process target attributes. For example, it adds "mtriple" and "mcpu" for NVPTX and ROCm target.

Reply:
In the future, we would love this preprocessor to deal with more complicated cases, for example, adding CUDA arch if it is not explicitly given, in order to make it more explicit to developers which arch is targeted at.